File: src/main/java/org/visual/view/MainLayout.java
package org.visual.view;

import io.avaje.inject.Lazy;
import jakarta.inject.Singleton;
import javafx.scene.control.Button;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.VBox;
import lombok.extern.slf4j.Slf4j;
import lombok.val;
import org.visual.component.GlobalMenu;

@Lazy
@Slf4j
@Singleton
public class MainLayout extends VBox {

  public MainLayout(GlobalMenu globalMenu) {
    val content =
        new BorderPane() {
          {
            val button = new Button("test");
            setCenter(button);
          }
        };
    content.setTop(globalMenu);
    getChildren().add(content);
  }
}

File: src/main/kotlin/org/visual/component/MainLayout.kt
package org.visual.component

import jakarta.inject.Inject
import jakarta.inject.Singleton
import javafx.scene.control.Button
import javafx.scene.control.ToolBar
import javafx.scene.layout.BorderPane
import javafx.scene.layout.VBox

@Singleton
class MainLayout @Inject constructor(globalMenu: GlobalMenu) : VBox() {

  private val content =
      BorderPane().apply {
        top = ToolBar().apply {}

        center = Button("test")
      }

  init {
    children.addAll(globalMenu, content)
  }
}
